src/app/Console/Commands/RequireDevTools.php

```diff
@@ -35,63 +35,71 @@ class RequireDevTools extends Command
      */
     public function handle()
     {
-        $this->progressBar = $this->output->createProgressBar(29);
+        $this->progressBar = $this->output->createProgressBar(45);
         $this->progressBar->minSecondsBetweenRedraws(0);
         $this->progressBar->maxSecondsBetweenRedraws(120);
         $this->progressBar->setRedrawFrequency(1);
-
         $this->progressBar->start();
 
         $this->info(' Requiring DevTools. Please wait...');
         $this->progressBar->advance();
 
-        // Check if auth exists
-        $details = null;
-        $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com']);
-        $process->run(function ($type, $buffer) use (&$details) {
-            if ($type !== Process::ERR && $buffer !== '') {
-                $details = json_decode($buffer);
-            } elseif (File::exists('auth.json')) {
-                $details = json_decode(File::get('auth.json'), true)['http-basic']['backpackforlaravel.com'] ?? false;
-            }
-            $this->progressBar->advance();
-        });
+        // Check if repositories exists
+        $this->composerRepositories();
 
-        // Create an auth.json file
-        if (! $details) {
-            $this->info(' Creating auth.json file with your authentication token');
+        // Check for authentication
+        $this->checkForAuthentication();
 
-            $this->line(' (Find your access token details on https://backpackforlaravel.com/user/tokens)');
-            $username = $this->ask('Access token username');
-            $password = $this->ask('Access token password');
+        // Require DevTools
+        $this->requireDevTools();
 
-            $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com', $username, $password]);
-            $process->run(function ($type, $buffer) use ($username, $password) {
-                if ($type === Process::ERR) {
-                    // Fallback
-                    $authFile = [
-                        'http-basic' => [
-                            'backpackforlaravel.com' => [
-                                'username' => $username,
-                                'password' => $password,
-                            ],
-                        ],
-                    ];
+        // Check if devtools is installed
+        $tries = 2;
+        while (! file_exists('vendor/backpack/devtools/src/Console/Commands/InstallDevTools.php')) {
```

Comment:
So this would just assume that if devtools install does not properly finish that the auth is incorrect right ? User might not have enough space and/or github is down.. whatever, we assume that is an authentication issue ? Or would the others fail with different errors before we get to the authentication issue ?

Reply:
@pxpm yes, I'm assuming any error is due to the authentication.
I'll change the code in order to detect the error, it will not be beautiful, because the errors come in the `$buffer` as text, I'll need to parse the string to figure out the error.

What should we do in case there are other errors?

Reply:
I am not sure, but I think that just output the error message should be enough, no matter what error it is right ? `Credentials dont match`, `not enough space`, `can't reach github` wtv ?

Reply:
Agree with @pxpm here. Or at least, if one message will be shown no matter what type of error happens, we should make the error message more general. Maybe something like `Could not authenticate those credentials. This could be due to incorrect credentials or a network error. If you're certain the credentials are correct, please try again later.`